app/selenium_app.py

```diff
@@ -92,14 +94,15 @@ def get_chunks_boundaries(data: Sized, desired_chunks_amount: int) -> Iterable[T
             yield i * chunk_size, data_size
 
 
-def get_element_id_to_is_displayed_mapping(page_content: str) -> Dict[str, bool]:
+def get_element_id_to_is_displayed_mapping(page_content: str, viewport: Dict) -> Dict[str, bool]:
     """Returns visibility status of all elements in the page
 
     Returned dictionary uses elements' jdn-hash property value as keys
     """
     escaped_page_content = str(page_content).encode('utf-8').decode('unicode_escape')
 
     driver = get_webdriver()
+    driver.set_window_size(viewport['width'], viewport['height'])
```

Comment:
I'm afraid window and viewport sizes are different. Viewport is just where page is rendered and window size includes toolbars/scrollbars if I'm not mistaken. I'd like to find a way to make it more precise

Reply:
Do you mean to calculate the entire window size, that will support desired size of the viewport?